data-protocols/dsp/dsp-catalog/dsp-catalog-http-dispatcher/src/main/java/org/eclipse/edc/protocol/dsp/catalog/dispatcher/delegate/CatalogRequestHttpDelegate.java

```diff
@@ -106,11 +110,18 @@ private String toJson(org.eclipse.edc.catalog.spi.protocol.CatalogRequestMessage
         try {
             var transformResult = transformerRegistry.transform(message, JsonObject.class);
             if (transformResult.succeeded()) {
-                return mapper.writeValueAsString(transformResult.getContent());
+                var compacted = compact(transformResult.getContent(), jsonLdContext());
+                return mapper.writeValueAsString(compacted);
             }
             throw new EdcException("Failed to write request.");
         } catch (JsonProcessingException e) {
             throw new EdcException("Failed to serialize catalog request", e);
         }
     }
+    
+    private JsonObject jsonLdContext() {
```

Comment:
Can you file an issue about the need to extract context creation as well? We will need to address this later.

Reply:
Created #2763